src/contexts/StateContextProvider.tsx

```diff
@@ -458,10 +459,23 @@ const StateContextProvider = ({children}: StateContextProviderProps) => {
     }, []);
 
     // Synchronize `logEventNumRef` with `logEventNum`.
+    // FIXME: is this necessary? Can we integrate it with the [numEvents, logEventNum] useEffect?
```

Comment:
Discussed offline and concluded that: it's fine to have a dedicated useEffect hook for syncing the variables.